src/libs/joystick/protocols/mavlink-manual-control.ts

```diff
@@ -502,16 +511,23 @@ export class MavlinkManualControlManager {
     // Re-use shift button if already mapped. Otherwise use R0 and S0.
     const regularShiftFunction = currentRegularButtonParameters.find((v) => v.actionId === MAVLinkButtonFunction.shift)
     const shiftActionValue = this.vehicleButtonParameterTable.find((e) => e.title === MAVLinkButtonFunction.shift)
-    if (regularShiftFunction === undefined) {
-      // Map shift to R0
-      this.vehicle.setParameter({ id: MAVLinkManualControlButton.R0, value: shiftActionValue!.value })
-
-      // Map shift to S0
-      this.vehicle.setParameter({ id: MAVLinkManualControlButton.S0, value: shiftActionValue!.value })
-    } else {
-      const sFunction = regularShiftFunction.button.replace('FUNCTION', 'SFUNCTION')
-      this.vehicle.setParameter({ id: sFunction, value: shiftActionValue!.value })
+    // use try catch to avoid undefined error
+
+    try {
```

Comment:
@rafaellehmkuhl without this try catch, the joystick module kept crashing on me.

Reply:
What error is happening? I think it would be good if we treat it directly instead of putting this entire code on a try/catch. We can be sweeping problems under the rug here.

As we didn't have complaints about it till now, I would bet they have something to do with the vehicle type?

Reply:
This PR isn't ment to go in, but mostly to review the changes I needed to make to get my setup working, this way we can come up with more permanent fixes together.  I'll have to get you the exact error I was getting later, but I believe the module was crashing on the setParameter call.

Reply:
without the try catch the controller stops working all together

Reply:
After further debugging it looks like this error is only present when I use my F310 joystick which I have attempted to add support for on my local copy. Still not sure what I did wrong with my F310 implementation, but I created PRs for my features that are independent of my F310 support